src/lib.rs

```diff
@@ -174,29 +174,35 @@ impl<'a> Chart<'a> {
         }
     }
 
-    /// Prints canvas content.
-    pub fn display(&mut self) {
+    pub fn to_string(&mut self) -> String {
+        let mut output = "".to_string();
```

Comment:
Instead of making a new `String` every frame, we could pre-allocate a string of just the right size as we know how big the canvas is. This would help live plotting performance. Thoughts?

Reply:
Refactored this function to completely remove the need for allocating any `String`